pkg/scalers/aws/aws_sigv4.go

```diff
@@ -0,0 +1,90 @@
+package aws
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"net/http"
+	"time"
+
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/service/amp"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	httputils "github.com/kedacore/keda/v2/pkg/util"
+)
+
+// Custom round tripper to sign requests
+type roundTripper struct {
+	client *amp.Client
+	region string
+}
+
+var (
+	// ErrAwsAMPNoAwsRegion is returned when "awsRegion" is missing from the config.
+	ErrAwsAMPNoAwsRegion = errors.New("no awsRegion given")
+)
+
+func (rt *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	cred, err := rt.client.Options().Credentials.Retrieve(req.Context())
+	if err != nil {
+		return nil, err
+	}
+	// Sign request
+	hasher := sha256.New()
+	reqCxt := v4.SetPayloadHash(req.Context(), hex.EncodeToString(hasher.Sum([]byte{})))
+	reqHash := v4.GetPayloadHash(reqCxt)
+	err = rt.client.Options().HTTPSignerV4.SignHTTP(req.Context(), cred, req, reqHash, "aps", rt.region, time.Now())
```

Comment:
I think that this will work too, and it avoid the requirement of using the hasher for hashing nothing
```suggestion
	// Sign request
	err = rt.client.Options().HTTPSignerV4.SignHTTP(req.Context(), cred, req, "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855", "aps", rt.region, time.Now())
```
Probably we could generate a const with the value and use it here for be clearer, but it can work.
I'd say that adding a comment claifying that "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855" is the sha256 of "" can be worth too

Reply:
no magic constants please :)

Reply:
do you prefer to set the value inline?

Reply:
when we introduce as below, the semgrp issue is

> // This is to avoid the semgrp issue of odd hash.Sum call flow when using hasher.sum(nil)
> 	// "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855" is the sha256 of ""
> 	const reqHash = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
> 
> 	err = rt.client.Options().HTTPSignerV4.SignHTTP(req.Context(), cred, req, reqHash, "aps", rt.region, time.Now())


Otherwise we could have it as above and ignore the semgrp issue

> hasher := sha256.New()
> reqCxt := v4.SetPayloadHash(req.Context(), hex.EncodeToString(hasher.Sum([]byte{})))
> reqHash := v4.GetPayloadHash(reqCxt)
> err =rt.client.Options().HTTPSignerV4.SignHTTP(req.Context(), cred, req, reqHash, "aps", rt.region, time.Now())

Please let me know the option to take here